iroh/src/commands/sync.rs

```diff
@@ -223,6 +269,74 @@ impl DocCommands {
                     println!("{}", fmt_entry(&entry));
                 }
             }
+            Self::Import {
+                doc,
+                author,
+                prefix,
+                path,
+                in_place,
+            } => {
+                let doc = get_doc(iroh, env, doc).await?;
+                let author = env.author(author)?;
+                let mut prefix = prefix.unwrap_or_else(|| String::from(""));
+
+                if prefix.ends_with('/') {
+                    prefix.pop();
+                }
+                let root = canonicalize_path(&path)?.canonicalize()?;
+                let tag = tag_from_file_name(&root)?;
+                let mut counts = (0, 0);
+                let stream = iroh
+                    .blobs
+                    .add_from_path(
+                        root.clone(),
+                        in_place,
+                        SetTagOption::Named(tag.clone()),
+                        WrapOption::NoWrap,
+                    )
+                    .await?;
+                let (_, _, entries) = aggregate_add_response(stream).await?;
+                let root_prefix = match root.parent() {
+                    Some(p) => p.to_path_buf(),
+                    None => PathBuf::new(),
+                };
+                for entry in entries.into_iter() {
+                    counts.0 += 1;
+                    counts.1 += entry.size;
+                    // adjust the key so that it does not leak the entire directory structure of
+                    // the importer's machine
+                    let key: Vec<u8> = PathBuf::from(entry.name.clone())
+                        .strip_prefix(root_prefix.clone())?
+                        .to_str()
+                        .map(|p| p.as_bytes())
+                        .ok_or(anyhow!("could not convert path to bytes"))?
+                        .into();
+                    doc.set_hash(author, key, entry).await?;
```

Comment:
sending the `entry` here, which contains the `Hash` and the size of the blob. This means we do not have to load the entry from the `bao_store` to discover the size. However, this also mean that you can "spoof" the document if you wanted to, and add a size that does not actually relate to the content from the hash.